### PR TITLE
WIP: Refactor

### PR DESCRIPTION
**Pull request by alimoghaddam2000**

### CODE REVIEW

File: src/graphics.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+int initialize_window(void)
+{
+    if (SDL_Init(SDL_INIT_EVERYTHING) != 0)
+    {
+        fprintf(stderr, "Error initializing SDL.\n");
+        return false;
+    }
+    window = SDL_CreateWindow(
+        "Snake!!",
```
Comment (ahmad-siavashi) [inline src/graphics.c]: It looks like the implementation is still aimed for Snake!

**Review by ahmad-siavashi — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+npC: src/forth.c
+	gcc -Wall -std=c99 src/forth.c src/gui.c src/logic.c src/graphics.c src/global.c src/utils.c src/stack.c `sdl2-config --libs --cflags` -lm -lncurses -o Final.out
+
```
Comment (ahmad-siavashi) [inline Makefile:3]: Usually `Final.out` is not something one would expect as the entry point of a program. If you changed it consider an update to the README.md file as well.

**Comment by ahmad-siavashi** [comment]
Excellent. You're ready to go and merge this PR. Update the document, i.e. README.md, if required.
Open more PRs for other changes in future.

**Comment by alimoghaddam2000** [comment]
Hi,
I hope this email finds you well. I wanted to express my gratitude for taking the time to review my code for the project. Your feedback and suggestions were incredibly valuable and have helped me improve my understanding of the concepts we have been learning.

I did update README.md and merge develop branch into main branch and I wanted to discuss the next steps for the project with you. I have a couple of ideas in mind and would appreciate your guidance on which direction to take.

Firstly, I was considering creating a wiki for my repository on GitHub. This would serve as a documentation resource for the project, providing explanations of the code structure, functionality, and usage instructions. It would be a valuable addition for future contributors or users of the project. I would like to know your thoughts on this and whether you believe it would be beneficial to proceed with creating the wiki.

Additionally, I have been working on implementing "auto completion" functionality in the project. This feature would enhance the user experience by suggesting possible commands or options as the user types. I am confident that this addition would greatly improve the usability of the project. Therefore, I would like to request a deadline for completing this task, so that I can allocate the necessary time and resources to finish it.

I would greatly appreciate your guidance and input on these next steps. Your expertise and advice have been instrumental in my progress, and I am eager to continue learning and improving.

Thank you once again for your time and support. I look forward to hearing from you soon.

Best regards,
Ali
________________________________

**Pull request merged**